packages/rs-drive-abci/src/execution/engine.rs

```diff
@@ -158,20 +162,72 @@ impl Platform {
                 .as_ref()
                 .map(|block_info| block_info.time_ms),
             proposer_pro_tx_hash: proposer,
+            proposed_app_version: proposed_version,
             validator_set_quorum_hash: Default::default(),
         };
 
         // println!("Block #{}", block_info.height);
 
-        let _block_begin_response = self
+        let block_begin_response = self
             .block_begin(block_begin_request, Some(&transaction))
-            .unwrap_or_else(|_| {
+            .unwrap_or_else(|e| {
                 panic!(
-                    "should begin process block #{} at time #{}",
+                    "should begin process block #{} at time #{} : {e}",
                     block_info.height, block_info.time_ms
                 )
             });
 
+        if block_begin_response.epoch_info.is_epoch_change
```

Comment:
I guess you need this code is running only in tests. I think we want this code in handlers which are called by JS Drive

Reply:
Moved to its own function.